src/chainparams.h

```diff
@@ -68,7 +68,7 @@ class CChainParams
     const std::vector<CDNSSeedData>& DNSSeeds() const { return vSeeds; }
     const std::vector<unsigned char>& Base58Prefix(Base58Type type) const { return base58Prefixes[type]; }
     const std::vector<CAddress>& FixedSeeds() const { return vFixedSeeds; }
-    virtual const Checkpoints::CCheckpointData& Checkpoints() const = 0;
```

Comment:
It had to be. Mainnet and testnet had different implementations for this method.

The result after this PR is obviously better, though.